python/src/aiconfig/editor/client/src/reducers/aiconfigReducer.ts

```diff
@@ -167,18 +167,27 @@ export default function aiconfigReducer(
       }));
     }
     case "UPDATE_PROMPT_MODEL": {
-      return reduceReplacePrompt(dirtyState, action.id, (prompt) => ({
-        ...prompt,
-        metadata: {
-          ...prompt.metadata,
-          model: action.modelName
-            ? {
-                name: action.modelName,
-                // TODO: Consolidate settings based on schema union
-              }
-            : undefined,
-        },
-      }));
+      return reduceReplacePrompt(dirtyState, action.id, (prompt) => {
+        // TODO: Consolidate settings based on schema union, server-side
+        // For now, just keep all the settings to match the server-side implementation
```

Comment:
I thought it was intentional to keep priority for client side over server (https://github.com/lastmile-ai/aiconfig/blob/05b10de06062876e8c91bbf60db38dcb16cbafa2/python/src/aiconfig/editor/client/src/reducers/aiconfigReducer.ts#L55-L56), but yea if we're updating the model action itself this makes sense to have server override:

Reply:
Ya, for the actions where we get the updated value on the server we want to have client values take precedent if they are values that can easily change quickly (input, settings). Probably part of this TODO will be implementing the logic on server and client, we just can't do it solely on the client right now because then we'd be in a weird state where the server has really different values in some cases